features/netsocket/TCPSocket.cpp

```diff
@@ -60,16 +63,15 @@ nsapi_error_t TCPSocket::connect(const SocketAddress &address)
         } else {
             blocking_connect_in_progress = true;
 
-            int32_t count;
+            uint32_t flag;
 
             // Release lock before blocking so other threads
             // accessing this object aren't blocked
             _lock.unlock();
-            count = _write_sem.wait(_timeout);
+            flag = _event_flag.wait(WRITE_FLAG, osFlagsWaitAny, _timeout);
             _lock.lock();
-
-            if (count < 1) {
-                // Semaphore wait timed out so break out and return
+            if (0 == (flag & WRITE_FLAG)) {
```

Comment:
_event_flag.wait returns 0xFFFFFFFEU on timeout, not zero (I don't really get it but it is what it is):
https://www.keil.com/pack/doc/CMSIS/RTOS2/html/group__flags__error__codes.html